### PR TITLE
Add a debug login API, add App\Utils\Date::getCarbon and add $event->last_cancel_date

### LINKED ISSUE #3: Implement an optional "last cancel date" for events
If you want to add this feature, here's what you should do:
* Add a column to `events` that will contain the last cancel date. This will require a migration
  * https://laravel.com/docs/6.x/migrations
* Edit `EventAdminController` to allow setting it in `store` and `update`
* Edit `resources/views/events/admin/{create,edit}.blade.php` to include this new row
  * Take a look at `resources/views/events/admin/regopts/{create,edit}.blade.php` for implementing datetime inputs
* Add logic to `EventPolicy` that will not allow `cancel`ing if last cancel date is in the past
* Be a nice person and add tests for this feature (#7)

### PR DESCRIPTION
**Pull request by Hegezcc**
Closes #3

### CODE REVIEW

**Comment by supertassu** [comment]
Thanks! I'll take a look at this later today.

File: resources/views/events/admin/edit.blade.php [hunk 1]
```diff
@@ -106,6 +106,31 @@
...
+        <label class="mt-6 block md:flex">
+            <div class="md:w-1/3">
+                <div class="font-semibold text-black mb-2">Last cancel date</div>
+                <div class="text-gray-700 text-sm">The last date attendees are able to cancel their spot. Not required.</div>
+            </div>
+
+            <div class="md:w-2/3">
+                <div>
+                    <input type="datetime-local" name="last_cancel_date" class="form-input w-full"
+                           required value="{{ old('last_cancel_date', \App\Utils\Date::fromString($event->last_cancel_date)->tz(config('app.timezone'))->format('Y-m-d\\TH:i')) }}">
```
Comment (supertassu) [inline resources/views/events/admin/edit.blade.php:118]: This causes `Call to a member function tz() on null` when editing an event where last cancel date is not set

File: routes/web.php [hunk 2]
```diff
@@ -11,6 +11,17 @@
 |
 */
 
+// Developer/testing helper routes
+
+if (env('APP_DEBUG')) {
```
Comment (supertassu) [inline routes/web.php:16]: This should be retrieved from config so it can be cached using config caching

File: routes/web.php [hunk 3]
```diff
@@ -11,6 +11,17 @@
...
+    Route::prefix('debug')->group(function () {
+        Route::get('login/{user_id}', function (int $user_id) {
+            Auth::login(App\Users\User::find($user_id));
```
Comment (supertassu) [inline routes/web.php:19]:
```suggestion
        Route::get('login/{user}', function (App\Users\User $user) {
            Auth::login($user);
```

File: routes/web.php (hunk 2 above)
Comment (Hegezcc) [reply routes/web.php]: Fixed in c8b76fa

File: resources/views/events/admin/edit.blade.php (hunk 1 above)
Comment (Hegezcc) [reply resources/views/events/admin/edit.blade.php]: Fixed in 86471e2

**Comment by Hegezcc** [comment]
Also fixed a logic bug where last_cancel_date would be required for canceling, or canceling wouldn't even be possible.

File: resources/views/events/admin/create.blade.php [hunk 4]
```diff
@@ -111,7 +111,8 @@
             <div class="md:w-2/3">
                 <div>
                     <input type="datetime-local" name="last_cancel_date" class="form-input w-full"
-                           required value="{{ old('last_cancel_date') }}">
+                           value="{{ old('last_cancel_date',
+                                $event->last_cancel_date ? \App\Utils\Date::fromString($event->last_cancel_date)->tz(config('app.timezone'))->format('Y-m-d\\TH:i') : null) }}">
```
Comment (supertassu) [inline resources/views/events/admin/create.blade.php:115]: Event is null as this is the `create` view

Comment (Hegezcc) [reply resources/views/events/admin/create.blade.php]: My bad as I just copied code without full manual testing, committed ae81471

**Comment by supertassu** [comment]
Tested and working. Many thanks!

**Pull request merged**